addons/sale/views/sale_views.xml

```diff
@@ -291,84 +320,171 @@
                                 </form>
-                                <tree string="Sales Order Lines" editable="bottom" decoration-info="invoice_status=='to invoice'">
-                                    <field name="sequence" widget="handle"/>
+                                <!-- This is inherited below to make the order lines non-editable (inline)
+                                    for the members of some usability groups (packaging, event):
+                                    Indeed in those cases we need a dialog because there are additional fields to input.
+                                -->
+                                <tree
+                                    string="Sales Order Lines"
+                                    editable="bottom"
+                                    decoration-info="(not display_type and invoice_status == 'to invoice')"
+                                >
+                                    <!-- if it is defined, it overrides the default button -->
+                                    <control>
+                                        <create string="Add a product"/>
+                                        <create string="Add a section" context="{'default_display_type': 'line_section'}"/>
+                                        <create string="Add a note" context="{'default_display_type': 'line_note'}"/>
+                                    </control>
+
+                                    <field name="sequence" widget="handle" />
+                                    <!-- We do not display the type because we don't want the user to be bothered with that information if he has no section or note. -->
+                                    <field name="display_type" invisible="1"/>
+
                                     <field name="product_updatable" invisible="1"/>
-                                    <field name="product_id"
-                                        attrs="{'readonly': [('product_updatable', '=', False)]}"
+                                    <field
+                                        name="product_id"
+                                        attrs="{
+                                            'readonly': [('product_updatable', '=', False)],
+                                            'required': [('display_type', '=', False)],
+                                        }"
                                         force_save="1"
-                                        context="{'partner_id':parent.partner_id, 'quantity':product_uom_qty, 'pricelist':parent.pricelist_id, 'uom':product_uom, 'company_id': parent.company_id,
-                                        'default_lst_price': price_unit, 'default_description_sale': name}"
-                                       />
-                                    <field name="layout_category_id" groups="sale.group_sale_layout"/>
+                                        context="{
```

Comment:
Still some unnecessary diff (not spotted everywhere, I think I wrote it sufficiently :p).